saleor/account/utils.py

```diff
@@ -31,6 +36,31 @@ def store_user_address(
             set_user_default_shipping_address(user, address)
 
 
+def is_user_address_limit_reached(user: "User"):
+    """Return True if user cannot have more addresses."""
+    return user.addresses.count() >= settings.MAX_USER_ADDRESSES
+
+
+def remove_the_oldest_user_address_if_address_limit_is_reached(user: "User"):
+    """Remove the oldest user address when max address limit is reached."""
+    if not is_user_address_limit_reached(user):
+        return
+
+    remove_the_oldest_user_address(user)
+
+
+def remove_the_oldest_user_address(user: "User"):
+    user_default_addresses_ids = [
+        user.default_billing_address_id,
+        user.default_shipping_address_id,
+    ]
+    user_address = (
+        user.addresses.exclude(pk__in=user_default_addresses_ids).order_by("pk").first()
+    )
+    if user_address:
+        user_address.delete()
```

Comment:
```suggestion
    user_address := (
        user.addresses.exclude(pk__in=user_default_addresses_ids).order_by("pk").first()
    ):
        user_address.delete()
```

Reply:
I'm not sure if it's more readable than the previous solution.